sympy/simplify/hyperexpand.py

```diff
@@ -1720,7 +1757,7 @@ def try_polynomial(func, z):
     if not al0:
         return None
 
-    a = al0[-1]
+    a = int(al0[-1]) # make int in case `a` is Float equal to int
```

Comment:
Could you find a test that passes through here and change one or more of its numerical values to Floats to see that this works?

Reply:
I'm unable to find such a case, but is that change required? The issue is fixed even without with change.

Reply:
If you can't find a test that passes through there then we don't even know it is tested. And if you do find one that passes through, I'm pretty sure I can cause it to fail by replacing ints with floats.

Reply:
Yes...and for `r` and `k`.

Reply:
I set up a trap for a test passing through the polynomial case.

Reply:
> Yes...and for `r` and `k`.

The test cases are not passing, also the OP issue now returns,     
```
  File "/home/praneeth/sympy/sympy/polys/polyutils.py", line 211, in _parallel_dict_from_expr_if_gens
    monom[indices[base]] = exp
KeyError: _t**4.0
```